src/reducers/importText.ts

```diff
@@ -157,22 +161,58 @@ const importText = (state: State, { path, text, lastUpdated, preventSetCursor, r
 
   }
   else {
+
     const json = isRoam ? roamJsonToBlocks(JSON.parse(convertedText)) : convertHTMLtoJSON(convertedText)
-    const imported = importJSON(state, simplePath, json, { lastUpdated, skipRoot })
 
-    return reducerFlow([
+    const uuid = createId()
 
-      updateThoughts(imported),
+    // Note: Create a dummy thought and then import new thoughts into its context and then collpase it . Since collapse uses existingThoughtMove it merges imported thoughts.
+    const updatedState = newThought(state, {
+      at: simplePath,
+      insertNewSubthought: true,
+      value: uuid
+    })
 
-      // restore the selection to the last imported thought on the first level
-      !preventSetCursor
-        ? setCursor({
-          path: imported.lastImported,
-          offset: headValue(imported.lastImported).length
-        })
-        : null,
+    const destEmpty = destThought.value === '' && getAllChildren(state, simpleContext).length === 0
 
-    ])(state)
+    // New dummy thoughts for collapsing
+    const newDummyThought = getAllChildren(updatedState, simpleContext).find(child => child.value === uuid)
+
+    const newDestinationPath = unroot(newDummyThought ? [...simplePath, newDummyThought] : simplePath) as SimplePath
+
+    const imported = importJSON(updatedState, newDestinationPath as SimplePath, json, { lastUpdated, skipRoot })
```

Comment:
`newDestinationPath` is already a `SimplePath` since you cast it on assignment, so you don't need the extra cast here.